pkg/storage/chunk/azure/blob_storage_client.go

```diff
@@ -68,8 +68,8 @@ var (
 					KeepAlive: 30 * time.Second,
 					DualStack: true,
 				}).Dial,
-				MaxIdleConns:           0,
-				MaxIdleConnsPerHost:    100,
+				MaxIdleConns:           512,
```

Comment:
Should these not all be configurable in S3/Azure Blob/GCS/etc? Seems like something we can move to the common storage config.

These changes might not be appropriate for everyone, and this is could be a fairly substantial change for some folks in some environments - not having a knob to turn would be unfortunate.

Reply:
After more thought I doubt this needs to be a knob, I'll reduce the value to 200 which seems to be what most OSS project are using.

If in the future someone needs to tune this, I'll happily add the config, but I'm not sure the effort is required right now.